src/middlewared/middlewared/plugins/virt/global.py

```diff
@@ -69,6 +69,19 @@ async def extend(self, data):
             data['state'] = Status.INITIALIZING.value
         return data
 
+    @private
+    async def license_active(self):
+        """
+        If this is iX enterprise hardware and has NOT been licensed to run virt plugin
+        then this will return False, otherwise this will return true.
+        """
+        can_run_virt = True
+        if await self.middleware.call('system.is_enterprise'):
+            license_ = await self.middleware.call('system.license')
+            can_run_virt = license_ is not None and 'JAILS' in license_['features']
```

Comment:
VMs are also licensed. You'll have to check which feature is licensed (jails and or VMs) and then check what the user is trying to create.